### ISSUE TITLE
[BUG] `save_changes()` doesn't throw an error when document doesn't exists

### ISSUE THREAD

**Issue by Luc1412** [roman-right/beanie#421]
**Describe the bug**
In 1.11.9 state was changed and always defaults being not None, instead it got a dict with the default values. This causes `check_if_state_saved` to never throw an error.

`save_changes` on a new created document that isn't saved in the database silently does nothing.

**To Reproduce**
```python
user_data = UserEntry(...)
user_data.x = ...
await user_data.save_changes()
```

**Expected behavior**
`StateNotSaved("No state was saved")`
**Additional context**
https://github.com/roman-right/beanie/compare/1.11.8...1.11.9
https://canary.discord.com/channels/822196934973456394/822196935435747332/1042243293662158970

**Comment by guesswh0** [comment]
This looks wrong when you expecting from `save_changes()` creating new documents.
But if document existed and you trying to save changes (e.g. some race condition deleted doc) this have to happen.

**Comment by Luc1412** [comment]
In my case I got a custom fetch document that either finds the document by _id or creates a new document if not found. 

To save changes, I only use save_changes() so I don't know weather the document is fetched from db or not. 

So I subclassed Document to overwrite save_changes() to insert the doc if it was just created.

Currently save() inserts the document too, if newly created, so it might make sense to implement that for save_changes() too. 
Otherwise at least something should happen when saving_changes on a non existent doc e.g. an exception should be raised.
